forte/data/extractor/predictor.py

```diff
@@ -0,0 +1,199 @@
+        super().initialize(resources, configs)
+        # TODO: load these params from outside
+        self.batch_size = 11
+        self.scope = Sentence
+
+        config1 = {
+            "scope": Sentence,
+            "entry": Token,
+            "vocab_use_unk": True
+        }
+        extractor1 = TextExtractor(config1)
+        converter1 = Converter(need_pad=True)
+
+        config2 = {
+            "scope": Sentence,
+            "entry": EntityMention,
+            "attribute": "ner_type",
+            "based_on": Token,
+            "strategy": "BIO",
+            "vocab_use_unk": True
+        }
+        extractor2 = AnnotationSeqExtractor(config2)
+        extractor2.add_entry(("PER", "B"))
+        converter2 = Converter(need_pad=True)
+
+        self.feature_scheme = {"text_tag": {
+                                    "extractor":  extractor1,
+                                    "converter": converter1
+                                },
+                                "ner_tag": {
+                                    "extractor":  extractor2,
+                                    "converter": converter2
+                                }
+                            }
+
+        class Model:
+            def __call__(self, tensor):
+                # Input shape: [Batch, length]
+                # Output shape: [Batch]
+                batch_size = len(tensor)
+                length = len(tensor[0])
+                return [[2 for _ in range(length)] for _ in range(batch_size)]
+        self.model = Model()
+
+    def convert(self, feature_collections):
+        example_collection = {}
+        for features_collection in feature_collections:
+            for tag, feature in features_collection.items():
+                if tag not in example_collection:
+                    example_collection[tag] = []
+                example_collection[tag].append(feature)
+
+        tensor_collection = {}
+        for tag, features in example_collection.items():
+            converter = self.feature_scheme[tag]["converter"]
+            tensor, mask = converter.convert(features)
+
+            if tag not in tensor_collection:
+                tensor_collection[tag] = {}
+
+            tensor_collection[tag]["tensor"] = tensor
+            tensor_collection[tag]["mask"] = mask
+        return tensor_collection
+
+    def yield_batch(self, pack):
+        for instance in pack.get(self.scope):
+            feature_collection = {}
+            for tag, scheme in self.feature_scheme.items():
+                extractor = scheme["extractor"]
+                feature = extractor.extract(pack, instance)
+                feature_collection[tag] = feature
+            
+            self.pack_pools.append(pack)
+            self.instance_pools.append(instance)
+            self.features_pools.append(feature_collection)
+
+            if len(self.features_pools) == self.batch_size:
```

Comment:
I guess you need to pass in an external Batcher here, in such case, you should let that batcher to design whether it is ready to yield.

Reply:
Ok. I will separate these functions into a new class "Batcher"

Reply:
Maybe discuss with Yanwen on which bather to use?